store/sqlstore/channel_store.go

```diff
@@ -2518,6 +2526,51 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 	return times, nil
 }
 
+func (s SqlChannelStore) CountUrgentPostsAfter(channelId string, timestamp int64, userId string) (int, error) {
+	joinLeavePostTypes := []string{
+		// These types correspond to the ones checked by Post.IsJoinLeaveMessage
+		model.PostTypeJoinLeave,
+		model.PostTypeAddRemove,
+		model.PostTypeJoinChannel,
+		model.PostTypeLeaveChannel,
+		model.PostTypeJoinTeam,
+		model.PostTypeLeaveTeam,
+		model.PostTypeAddToChannel,
+		model.PostTypeRemoveFromChannel,
+		model.PostTypeAddToTeam,
+		model.PostTypeRemoveFromTeam,
+	}
+
+	query := s.getQueryBuilder().
+		Select("count(*)").
+		From("Posts").
+		Join("PostsPriority ON Posts.Id = PostsPriority.PostId").
+		Where(sq.And{
+			sq.Eq{"Posts.ChannelId": channelId},
+			sq.Gt{"Posts.CreateAt": timestamp},
+			sq.NotEq{"Posts.Type": joinLeavePostTypes},
+			sq.Eq{"Posts.DeleteAt": 0},
+			sq.Eq{"PostsPriority.Priority": model.PostPropsPriorityUrgent},
+		})
+
+	if userId != "" {
+		query = query.Where(sq.Eq{"UserId": userId})
+	}
+
+	var urgent int64
+	sql, args, err := query.Where(sq.Eq{"RootId": ""}).ToSql()
```

Comment:
nit: let's include this `Where` condition above, and have a dedicated line for `query.ToSql`.